lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -303,6 +304,30 @@ def __init__(self, module):
         self._ensure_dnf()
         self.lockfile = "/var/cache/dnf/*_lock.pid"
         self.pkg_mgr_name = "dnf"
+        self.base = self._base()
+        self.exclude_pkgnames = self._get_exclude_pkgnames()
+
+    def _pattern_to_pkgname(self, base, pattern):
+        if dnf.util.is_glob_pattern(pattern):
+            q = base.sack.query().filter(name__glob=pattern)
+            return map(lambda p: p.name, q)
```

Comment:
Anytime I reach for map or filter, I like to use a list comprehension of generator exception instead as it is faster and considered more pythonic.
``` python
return [p.name for p in q]
```